pkg/cloud/cloud.go

```diff
@@ -566,9 +586,16 @@ func execBatchDescribeVolumes(svc util.EC2API, input []string, batcher volumeBat
 
 	resp, err := describeVolumes(ctx, svc, request)
 	if err != nil {
+		volumeIDRegex := regexp.MustCompile("vol-[a-zA-Z0-9]+")
+		badVolumes := volumeIDRegex.FindAllString(err.Error(), -1)
+		for _, volID := range badVolumes {
+			c.badVolumeIDs.Set(volID, aws.Bool(true))
+		}
 		return nil, err
 	}
 
+	retryBadVolumes(c, badVolumes, svc)
```

Comment:
Given this is only used in one place, does a separate function really make sense?
Also, you need to not discard the response(s) if the call succeeds.